### PR TITLE
Feature/socket

### PR DESCRIPTION
**Pull request by NChitty**
Ready for review

I could not find a way to create the client instance and send it to the front end from the backend, but I can look into it

### CODE REVIEW

**Comment by suneettipirneni** [comment]
I would keep this PR as a draft PR instead of having it open since it isn't feature complete.

**Comment by Armintron** [comment]
This PR is now ready for review.
Frontend implementation skeleton code:
```js
  import { io } from 'socket.io-client';
  useEffect(() => {
    // set up socket instance
    const socket = io(BACKEND URL);
    // add event listeners once the socket connects to the backend
    socket.on('connect', () => {
      // tell the server which groups you want to recieve events for.
      // think of this as joining a "room" for each group.
      // all sockets that are specific to a group have a groupId passed to them-
      // to show which group the event was emitted to.
      // all sockets below, for now, will only listen to joined groups.
      // sockets implemented in the future may not be this way, depending on their purpose.
      socket.emit('join', ['groupIdsToRecieveEventsFrom']);
      // when an image is uploaded to a joined group
      // the uploaded image and groupId uploaded to are passed in
      socket.on('image uploaded', (image, groupId) => {
        console.log(`image uploaded to group ${groupId}`);
        console.log('Image data:');
        console.log(image);
        /* do stuff */
      });
      // when users join a group
      // the amount of users that joined the group given by groupId are passed in 
      socket.on('users joined', (joinCount, groupId) => {
        console.log(`${joinCount} users joined group ${groupId}`);
        /* do stuff */
      });
      // when users leave a group
      // the amount of users that left the group given by groupId are passed in 
      socket.on('users removed', (removeCount, groupId) => {
        console.log(`${removeCount} users left group ${groupId}`);
        /* do stuff */
      });
    });
  }, []);
```
The use of `useEffect` here was my attempt at making sure this code only runs once when the page loads. It is NOT important that you specifically have this code in `userEffect`, but it IS important you make sure the code only runs once on page load, or else you may receive duplicate event triggers for a single event.

File: package.json [hunk 1]
```diff
@@ -42,6 +42,8 @@
     "pino": "^6.11.1",
     "pino-pretty": "^4.6.0",
     "sharp": "^0.28.0",
+    "socket.io": "^4.0.1",
+    "socket.io-client": "^4.0.1",
```
Comment (suneettipirneni) [inline package.json:46]: I'm guessing the client package isn't needed on the server 👀

**Review by suneettipirneni — APPROVED** [review]
lgtm

**Pull request merged**